_posts/2024-09-26-managing-connector-offsets.md

```diff
@@ -0,0 +1,252 @@
+    kind: KafkaConnector
+    name: my-source-connector
+    uid: 637e3be7-bd96-43ab-abde-c55b4c4550e0
+  resourceVersion: "66951"
+  uid: 641d60a9-36eb-4f29-9895-8f2c1eb9638e
+data:   (3)
+  offsets.json: |-
+    {
+      "offsets" : [ {
+        "partition" : {
+          "filename" : "/opt/kafka/LICENSE"
+        },
+        "offset" : {
+          "position" : 15295
+        }
+      } ]
+    }
+```
+
+1. If the ConfigMap doesn't already exist Strimzi will create it.
+2. The owner reference points to your KafkaConnector resource. To provide a custom owner reference, create the ConfigMap in advance and set an owner reference.
+3. Strimzi puts the offsets into a field called `offsets.json`. It doesn't overwrite any other fields when updating an existing ConfigMap.
+
+You can check that the output matches the results from the Connect REST API by calling the `GET /connectors/{connector}/offsets` endpoint directly:
+
+```shell
+$ kubectl exec -n kafka -it my-connect-cluster-connect-0 -- curl localhost:8083/connectors/my-source-connector/offsets
+{"offsets":[{"partition":{"filename":"/opt/kafka/LICENSE"},"offset":{"position":15295}}]}
+```
+
+### Altering offsets
+
+To alter offsets for a connector you also need a ConfigMap, this time to tell Strimzi the new offsets to set.
+All sink connectors use the same format, however for source connectors it varies.
+The easiest way to create the ConfigMap for altering offsets, is actually to reuse the ConfigMap that Strimzi wrote the offsets into.
+
+To alter connector offsets the connector also needs to be stopped.
+
+Edit the `KafkaConnector` resource to set the `my-connector-offsets` ConfigMap as the source of offsets for the alter operation, and set the `state` as `stopped`:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaConnector
+# ...
+spec:
+  #...
+  state: stopped
+  alterOffsets:
+    fromConfigMap:
+      name: my-connector-offsets
+  #...
+```
+
+The `status` in the `KafkaConnector` is updated once Strimzi has stopped the connector.
+
+Now edit the `my-connector-offsets` ConfigMap to change the `position` to, for example, 10:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+# ...
+data:
+  offsets.json: |-
+    {
+      "offsets" : [ {
+        "partition" : {
+          "filename" : "/opt/kafka/LICENSE"
+        },
+        "offset" : {
+          "position" : 10
+        }
+      } ]
+    }
+```
+
+Finally trigger the operator to alter offsets by annotating the resource:
+
+```shell
+$ kubectl annotate kafkaconnector my-source-connector strimzi.io/connector-offsets=alter -n kafka
+```
+
+Strimzi removes the `strimzi.io/connector-offsets` annotation from the resource once the offsets have been successfully updated.
+You can also verify this directly:
+
+```shell
+$ kubectl exec -n kafka -it my-connect-cluster-connect-0 -- curl localhost:8083/connectors/my-source-connector/offsets
+{"offsets":[{"partition":{"filename":"/opt/kafka/LICENSE"},"offset":{"position":10}}]}
+```
+
```

Comment:
I think it would make sense to add a chapter here that user will need to _unstopp_ the connector again by editing the CR again as it might not be obvious that it doesn't happen automatically.